The job **`i686-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/5366993848/jobs/9736792611) [(plain)](https://github.com/rust-lang-ci/rust/commit/b716bc479aaeb4143d9f05a9390fb9d659d57525/checks/14528279427/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    found:    334d016f755cd6dc58c53a86e183882f8ec14f52fb05345887c8a5edd42c87b7
    expected: 64ec88ca00b268e5ba1a35678a1b5316d212f4f366b2477232534a8aeca37f3c
..
======================================================================
ERROR: test_cargoflags (bootstrap_test.BuildBootstrap.test_cargoflags)
Traceback (most recent call last):
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 138, in test_cargoflags
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 138, in test_cargoflags
    args, _ = self.build_args(env={"CARGOFLAGS": "--timings"})
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 135, in build_args
configure: build.configure-args := []
configure: profile              := user
configure: build.configure-args := ['--set', 'rust.deny-warnings=false']
configure: build.configure-args := ['--set', 'rust.deny-warnings=false']
configure: rust.deny-warnings   := False
configure: profile              := user
configure: build.configure-args := []
    return build.build_bootstrap_cmd(env), env
  File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 943, in build_bootstrap_cmd
  File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 943, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `C:\a\rust\rust\build\x86_64-pc-windows-gnu\stage0\bin\cargo.exe`
======================================================================
======================================================================
ERROR: test_warnings (bootstrap_test.BuildBootstrap.test_warnings)
Traceback (most recent call last):
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 147, in test_warnings
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 147, in test_warnings
    _, env = self.build_args(configure_args, args=["--warnings=warn"])
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 135, in build_args
  File "C:\a\rust\rust\src\bootstrap\bootstrap_test.py", line 135, in build_args
    return build.build_bootstrap_cmd(env), env
  File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 943, in build_bootstrap_cmd
  File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 943, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `C:\a\rust\rust\build\x86_64-pc-windows-gnu\stage0\bin\cargo.exe`
----------------------------------------------------------------------
Ran 12 tests in 0.621s

FAILED (errors=2)
---
configure: profile              := user
configure: build.configure-args := ['--set', 'profile=compiler']
configure: profile              := compiler
Build completed unsuccessfully in 1:12:03
make: *** [Makefile:66: ci-msvc-ps1] Error 1
```

</details>

--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -1046,6 +1046,8 @@ def bootstrap(args):
     if not using_default_path or os.path.exists(toml_path):
         with open(toml_path) as config:
             config_toml = config.read()
+    else:
+        config_toml = ''
 
     profile = RustBuild.get_toml_static(config_toml, 'profile')
     if profile is not None:

--- a/src/bootstrap/bootstrap_test.py
+++ b/src/bootstrap/bootstrap_test.py
@@ -132,6 +132,7 @@ class BuildBootstrap(unittest.TestCase):
         parsed = bootstrap.parse_args(args)
         build = serialize_and_parse(configure_args, parsed)
         build.build_dir = os.environ["BUILD_DIR"]
+        build.build = os.environ["BUILD_PLATFORM"]
         return build.build_bootstrap_cmd(env), env
 
     def test_cargoflags(self):

--- a/src/bootstrap/configure.py
+++ b/src/bootstrap/configure.py
@@ -400,7 +400,9 @@ def parse_example_config(known_args, config):
     targets = {}
     top_level_keys = []
 
-    for line in open(rust_dir + '/config.example.toml').read().split("\n"):
+    with open(rust_dir + '/config.example.toml') as example_config:
+        example_lines = example_config.read().split("\n")
+    for line in example_lines:
         if cur_section is None:
             if line.count('=') == 1:
                 top_level_key = line.split('=')[0]

--- a/src/bootstrap/test.rs
+++ b/src/bootstrap/test.rs
@@ -2667,6 +2667,7 @@ impl Step for Bootstrap {
         check_bootstrap
             .args(["-m", "unittest", "bootstrap_test.py"])
             .env("BUILD_DIR", &builder.out)
+            .env("BUILD_PLATFORM", &builder.build.build.triple)
             .current_dir(builder.src.join("src/bootstrap/"))
             .args(builder.config.test_args());
         try_run(builder, &mut check_bootstrap).unwrap();

